### DISCUSSION TITLE
How to clean already played audio files ?

### DISCUSSION THREAD

**Discussion by jooola** [savonet/liquidsoap#2601] [Q&A] [answered]
In need to clean a cache directory filled with files liquidsoap is playing or has played.

The files are queued using `request.queue`.

I was wondering if liquidsoap shipped a nice way to get this info ?

I can use `lsof` for this, but I was hoping that liquidsoap could have a function for this ?

**Comment by smimram** [comment, marked as answer]
If I understand correctly, you want to remove files which have already been played? In this case, the best is to pass `temporary=true` when creating the request: the file will then be removed once the request has been played. Something like
```
q = request.queue()
...
r = request.create(temporary=true, "/tmp/filename.mp3")
q = push(r)
```
Otherwise, please be more specific about what you have in mind.

**Reply by jooola** [reply]
Sorry for the lake of details. But you have answered my question !

What will happen if the same file is reused in another future request ? How will the race condition be handled in that case ?

**Reply by smimram** [reply]
I would advise against using the same file twice: the first request will remove the file when being distroyed, which will cause problems when the second one is played I guess...
